cirq/protocols/has_stabilizer_effect_protocol.py

```diff
@@ -59,3 +61,16 @@ def _strat_has_stabilizer_effect_from_gate(val: Any) -> Optional[bool]:
     if hasattr(val, 'gate'):
         return _strat_has_stabilizer_effect_from_has_stabilizer_effect(val.gate)
     return None
+
+
+def _strat_has_stabilizer_effect_from_unitary(val: Any) -> Optional[bool]:
+    """Attempts to infer whether val has stabilizer effect from its unitary.
+
```

Comment:
Also add that this is constrained to 2x2 unitaries.

Reply:
Done.